src/persist-client/src/fetch.rs

```diff
@@ -466,12 +466,14 @@ where
         //   these batches, all data physically in the batch but outside of the
         //   truncated bounds must be ignored. Not every user batch is
         //   truncated.
-        // - Batches written by compaction. These always have an inline desc
-        //   that exactly matches the one they are registered with. The since
-        //   can be anything.
+        // - Batches written by compaction. These always have an inline desc that
```

Comment:
hmmm i'm not sure i understand why the lower is guaranteed to be the same. feels like it could be any subset, no? like i'd expect `[one run, empty, empty]`, `[empty, one run, empty]`, `[empty, empty, one run]` to all get the special case

Reply:
not guaranteed to be the same, just didn't test adequately!